### PR TITLE
Fix issue 47: Upgraded Packages to v0.8.1 of operator-sdk

### PR DESCRIPTION
**Pull request by AdlerFleurant**
The current version of the cert-operator can no longer compiles against the latest version of the operator-sdk. As it is currently set to master, this will upgrade all required packages to the latest version for the operator-sdk.

### CODE REVIEW

**Comment by mathianasj** [comment]
Would it be better to lock to a version of the sdk or always use the latest and risk breaking changes?  @etsauer thoughts?

**Comment by AdlerFleurant** [comment]
I agree that this would help a lot. We should jump from published stable version to published stable version. 
e2e tests pass for this.

File: Gopkg.toml [hunk 1]
```diff
@@ -65,24 +50,26 @@ required = [
   name = "github.com/operator-framework/operator-sdk"
   # The version rule is used for a specific release and the master branch for in between releases.
   branch = "master" #osdk_branch_annotation
```
Comment (mathianasj) [inline Gopkg.toml]: this is actually using the master branch and not the declared commented version of v0.8.0.  Is this intended?

File: Gopkg.toml [hunk 2]
```diff
@@ -1,37 +1,18 @@
...
-  revision = "0cf8f7e6ed1d2e3d47d02e3b6e559369af24d803"
-
 [[override]]
   name = "github.com/go-openapi/spec"
   branch = "master"
 
 [[override]]
   name = "sigs.k8s.io/controller-tools"
-  version = "=v0.1.8"
+  revision = "9d55346c2bde73fb3326ac22eac2e5210a730207"
```
Comment (mathianasj) [inline Gopkg.toml]: If i remember correctly this is to provide a bug fix that is not in a tagged release yet?

File: cmd/manager/main.go [hunk 3]
```diff
@@ -115,17 +125,50 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Create Service object to expose the metrics port.
-	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	if err = serveCRMetrics(cfg); err != nil {
```
Comment (mathianasj) [inline cmd/manager/main.go]: thanks this would have saved me a lot of debugging time

File: Gopkg.toml (hunk 1 above)
Comment (AdlerFleurant) [reply Gopkg.toml]: This is intended as this update is relying on the master branch. I followed the trend where the operator-sdk is relying on master. My thinking was that once, the next version of the sdk is published we would published freeze these dependencies and published a new version of the cert-operator.

File: Gopkg.toml (hunk 2 above)
Comment (AdlerFleurant) [reply Gopkg.toml]: That is true and it is still the case now. So we have to rely on it until resolved.

File: cmd/manager/main.go (hunk 3 above)
Comment (AdlerFleurant) [reply cmd/manager/main.go]: operator-sdk now generates this code for you.

File: Gopkg.toml (hunk 1 above)
Comment (AdlerFleurant) [reply Gopkg.toml]: Let's see what @etsauer thoughts on this. And we can keep it as is or freeze the version now 0.8.1 right?

Comment (AdlerFleurant) [reply Gopkg.toml]: It has been decided to freeze on 0.8.1 instead.

File: Gopkg.toml (hunk 2 above)
Comment (AdlerFleurant) [reply Gopkg.toml]: controller-tools is now locked on the version recommended by operator-sdk v0.8.1

**Comment by AdlerFleurant** [comment]
@etsauer @mathianasj Completed with fixing the change to v0.8.1 of the operator-sdk

**Comment by etsauer** [comment]
@AdlerFleurant got the following when running the tests.

```
time="2019-07-08T21:59:20-04:00" level=info msg="Local operator stderr: {\"level\":\"info\",\"ts\":1562637307.168744,\"logger\":\"cmd\",\"msg\":\"Go Version: go1.10.8\"}\n{\"level\":\"info\",\"ts\":1562637307.1687684,\"logger\":\"cmd\",\"msg\":\"Go OS/Arch: linux/amd64\"}\n{\"level\":\"info\",\"ts\":1562637307.1687737,\"logger\":\"cmd\",\"msg\":\"Version of operator-sdk: v0.8.1\"}\ntime=\"2019-07-08T21:55:07-04:00\" level=info msg=\"Loading default config file from /etc/cert-operator/config.yaml\"\n{\"level\":\"info\",\"ts\":1562637307.173064,\"logger\":\"leader\",\"msg\":\"Trying to become the leader.\"}\n{\"level\":\"info\",\"ts\":1562637307.1730921,\"logger\":\"leader\",\"msg\":\"Skipping leader election; not running in a cluster.\"}\n{\"level\":\"info\",\"ts\":1562637309.4757297,\"logger\":\"cmd\",\"msg\":\"Registering Components.\"}\n{\"level\":\"info\",\"ts\":1562637309.4759798,\"logger\":\"controller_route\",\"msg\":\"SSL Not Verified\"}\n{\"level\":\"info\",\"ts\":1562637309.4760087,\"logger\":\"controller_route\",\"msg\":\"None provider.\"}\n{\"level\":\"info\",\"ts\":1562637309.4761286,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting EventSource\",\"controller\":\"route-controller\",\"source\":\"kind source: /, Kind=\"}\n{\"level\":\"info\",\"ts\":1562637309.4764998,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting EventSource\",\"controller\":\"route-controller\",\"source\":\"kind source: /, Kind=\"}\n{\"level\":\"info\",\"ts\":1562637309.4767866,\"logger\":\"controller_service\",\"msg\":\"SSL Not Verified\"}\n{\"level\":\"info\",\"ts\":1562637309.4768014,\"logger\":\"controller_service\",\"msg\":\"None provider.\"}\n{\"level\":\"info\",\"ts\":1562637309.476871,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting EventSource\",\"controller\":\"service-controller\",\"source\":\"kind source: /, Kind=\"}\n{\"level\":\"info\",\"ts\":1562637309.4771836,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting EventSource\",\"controller\":\"service-controller\",\"source\":\"kind source: /, Kind=\"}\n{\"level\":\"info\",\"ts\":1562637311.4332814,\"logger\":\"metrics\",\"msg\":\"Skipping metrics Service creation; not running in a cluster.\"}\n{\"level\":\"info\",\"ts\":1562637311.433383,\"logger\":\"cmd\",\"msg\":\"Starting the Cmd.\"}\n{\"level\":\"info\",\"ts\":1562637311.7338722,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting Controller\",\"controller\":\"route-controller\"}\n{\"level\":\"info\",\"ts\":1562637311.7338111,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting Controller\",\"controller\":\"service-controller\"}\n{\"level\":\"info\",\"ts\":1562637311.8343334,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting workers\",\"controller\":\"service-controller\",\"worker count\":1}\n{\"level\":\"info\",\"ts\":1562637311.8343332,\"logger\":\"kubebuilder.controller\",\"msg\":\"Starting workers\",\"controller\":\"route-controller\",\"worker count\":1}\n{\"level\":\"info\",\"ts\":1562637311.8347971,\"logger\":\"controller_route\",\"msg\":\"Reconciling Route\",\"Request.Namespace\":\"cert-operator-test\",\"Request.Name\":\"route-tls\"}\n{\"level\":\"info\",\"ts\":1562637311.906184,\"logger\":\"controller_route\",\"msg\":\"Updated route with new certificate\",\"Request.Namespace\":\"cert-operator-test\",\"Request.Name\":\"route-tls\"}\n{\"level\":\"info\",\"ts\":1562637314.237431,\"logger\":\"controller_service\",\"msg\":\"Reconciling Service\",\"Request.Namespace\":\"cert-operator-test\",\"Request.Name\":\"example-service\"}\n{\"level\":\"info\",\"ts\":1562637314.3863537,\"logger\":\"controller_service\",\"msg\":\"Updated service with new certificate\",\"Request.Namespace\":\"cert-operator-test\",\"Request.Name\":\"example-service\"}\n{\"level\":\"info\",\"ts\":1562637319.371303,\"logger\":\"controller_service\",\"msg\":\"Reconciling Service\",\"Request.Namespace\":\"cert-operator-test\",\"Request.Name\":\"example-service-pkcs12\"}\nE0708 21:55:19.371662   14435 runtime.go:69] Observed a panic: \"invalid memory address or nil pointer dereference\" (runtime error: invalid memory address or nil pointer dereference)\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51\n/usr/local/go/src/runtime/asm_amd64.s:573\n/usr/local/go/src/runtime/panic.go:502\n/usr/local/go/src/runtime/panic.go:63\n/usr/local/go/src/runtime/signal_unix.go:388\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/pkg/controller/service/service_controller.go:152\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\n/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\n/usr/local/go/src/runtime/asm_amd64.s:2361\npanic: runtime error: invalid memory address or nil pointer dereference [recovered]\n\tpanic: runtime error: invalid memory address or nil pointer dereference\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x11e89f7]\n\ngoroutine 330 [running]:\ngithub.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107\npanic(0x13e7980, 0x20994e0)\n\t/usr/local/go/src/runtime/panic.go:502 +0x229\ngithub.com/redhat-cop/cert-operator/pkg/controller/service.(*ReconcileService).Reconcile(0xc420a8a3c0, 0xc42080c500, 0x12, 0xc42080c460, 0x16, 0x20b42c0, 0x0, 0x0, 0x0)\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/pkg/controller/service/service_controller.go:152 +0xc17\ngithub.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc420840460, 0x0)\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x188\ngithub.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36\ngithub.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4202900a0)\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54\ngithub.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4202900a0, 0x3b9aca00, 0x0, 0x1, 0xc4205be2a0)\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd\ngithub.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4202900a0, 0x3b9aca00, 0xc4205be2a0)\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d\ncreated by github.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start\n\t/home/esauer/go/src/github.com/redhat-cop/cert-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x35b\n"
```

**Comment by etsauer** [comment]
@AdlerFleurant looking more at this... it appears that the nil pointer panic is coming from this line:

https://github.com/AdlerFleurant/cert-operator/blob/Fix_Issue_47/pkg/controller/service/service_controller.go#L151

```
...
runtime error: invalid memory address or nil pointer dereference
...
/home/esauer/go/src/github.com/redhat-cop/cert-operator/pkg/controller/service/service_controller.go:151
```

**Comment by etsauer** [comment]
@AdlerFleurant I had local config files that were messing up the tests. Cleaning them up fixed the build.

**Comment by AdlerFleurant** [comment]
This also fixed #15

**Pull request merged**